internal/gcsx/inactive_timeout_reader_test.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcsx
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/clock"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/fake"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/gcs"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type InactiveTimeoutReaderTestSuite struct {
+	suite.Suite
+	ctx        context.Context
+	mockBucket *storage.TestifyMockBucket
+	object     *gcs.MinObject
+	reader     gcs.StorageReader // The reader under test
+
+	// Fields to hold results from setup for individual tests
+	initialData       []byte
+	readHandle        []byte
+	initialFakeReader *fake.FakeReader
+	timeout           time.Duration
+	simulatedClock    *clock.SimulatedClock
+}
+
+func TestInactiveTimeoutReaderTestSuite(t *testing.T) {
+	suite.Run(t, new(InactiveTimeoutReaderTestSuite))
+}
+
+func (s *InactiveTimeoutReaderTestSuite) SetupTest() {
```

Comment:
[Avoid suites if possible.](https://google.github.io/styleguide/go/best-practices.html#keep-setup-code-scoped-to-specific-tests)